### PR TITLE
Add default config for Gretel GPT with differential privacy

### PR DESCRIPTION
**Pull request by zredlined**
Add dedicated configuration file for differentially private fine-tuning of language models on natural language data. 
- Added top-level comments explaining the config's purpose and key considerations (e.g. 10k records+ recommended)
- Using optimized parameters for DP defaults
- Inline comments for important parameters

### CODE REVIEW

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+schema_version: "1.0"
+name: "natural-language-dp"
+models:
+  - gpt_x:
+      data_source: "__temp__" 
+      pretrained_model: "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+      column_name: null  # Specify column name for data if using multiple columns
+      params:
+        batch_size: 16  # Number of samples processed before the model is updated
```
Comment (andreatgretel) [inline config_templates/gretel/synthetics/natural-language-differential-privacy.yml:19]: We have now exposed `gradient_accumulation_steps` (which before was always 8 by default), so the "effective batch size" is `batch_size * gradient_accumulation_steps`. Should we maybe have that parameter added here as well?

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        steps: 3000  # Total number of training steps
+        weight_decay: 0.01
+        warmup_steps: 100
+        lr_scheduler: "linear"
+        learning_rate: 0.0001 # Initial learning rate for training
```
Comment (andreatgretel) [inline config_templates/gretel/synthetics/natural-language-differential-privacy.yml:24]: In my experience, e.g. for experiments we did for the blog post and others, we typically have better results with 0.001, maybe we can make that the default? @lipikaramaswamy might have a different opinion

**Review by andreatgretel — APPROVED** [review]

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+schema_version: "1.0"
+name: "natural-language-dp"
+models:
+  - gpt_x:
+      data_source: "__temp__" 
+      pretrained_model: "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+      column_name: null  # Specify column name for data if using multiple columns
+      params:
+        batch_size: 16  # Number of samples processed before the model is updated
+        steps: 3000  # Total number of training steps
```
Comment (lipikaramaswamy) [inline config_templates/gretel/synthetics/natural-language-differential-privacy.yml:20]: IMO a better param to set for DP FT is `epochs: 3`. 3k steps could be overkill or insufficient depending on dataset size.

Comment (lipikaramaswamy) [reply config_templates/gretel/synthetics/natural-language-differential-privacy.yml:20]: e.g. for the recommended min 10k dataset size and effective batch size of `16 * 8` , we'd end up with 78 steps per epoch, so 3k steps would be 38 epochs!

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml (hunk 2 above)
Comment (lipikaramaswamy) [reply config_templates/gretel/synthetics/natural-language-differential-privacy.yml:24]: Our experiments were with `mistralai/Mistral-7B-Instruct-v0.2`, and we found learning rates in the range `0.00095 - 0.0005` produced better results. I noticed [some runs](https://console.gretel.ai/proj_2ieHRNEQZdmEj4E0FQmFxkPMxLI/models/6683558dceef6c9ca0fe25f3/config) with TinyLlama and `learning_rate: 0.001` producing good quality results too. 

I would agree that `0.001` is a good default for this blueprint.

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        lr_scheduler: "linear"
+        learning_rate: 0.0001 # Initial learning rate for training
+        max_tokens: 128  # Increase to allow for longer sequences
+      peft_params:
+        lora_r: 8
+        lora_alpha_over_r: 1
+      privacy_params:
+        dp: true  # Enable differentially private fine-tuning via DP-SGD
+        epsilon: 5  # Privacy budget (lower values = stronger privacy)
+        delta: auto  # Probability of privacy loss (auto-calculated)
```
Comment (lipikaramaswamy) [inline config_templates/gretel/synthetics/natural-language-differential-privacy.yml:32]: Suggest `delta: auto # Probability of privacy leakage (auto-calculated)`

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml (hunk 1 above)
Comment (zredlined) [reply config_templates/gretel/synthetics/natural-language-differential-privacy.yml:19]: @andreatgretel Added this to "key points" and the extra param

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml (hunk 2 above)
Comment (zredlined) [reply config_templates/gretel/synthetics/natural-language-differential-privacy.yml:24]: Sounds great, we'll go with 0.001

File: config_templates/gretel/synthetics/natural-language-differential-privacy.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+schema_version: "1.0"
+name: "natural-language-dp"
+models:
+  - gpt_x:
+      data_source: "__temp__" 
+      pretrained_model: "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+      column_name: null  # Specify column name for data if using multiple columns
+      params:
+        batch_size: 16  # Number of samples processed before the model is updated
```
Comment (andreatgretel) [inline config_templates/gretel/synthetics/natural-language-differential-privacy.yml:23]: "Number of samples used to compute gradient" maybe?

**Review by lipikaramaswamy — APPROVED** [review]
